lib/matplotlib/tests/test_mlab.py

```diff
@@ -314,122 +208,107 @@ def test_stride_ensure_integer_type(self):
-                    datetime.datetime(1976, 3, 5, 0, 0, 1),
-                    datetime.datetime(1983, 7, 9, 17, 17, 34),
-                    datetime.datetime(2054, 6, 20, 14, 31, 45),
-                    datetime.datetime(2000, 10, 31, 11, 50, 23)]
-        self.fd.seek(0)
-        array = mlab.csv2rec(self.fd, names='a', dayfirst=True)
-        assert_array_equal(array['a'].tolist(), expected)
-
-    def test_csv2rec_yearfirst(self):
-        self.fd.write('14/01/11\n'
-                      '76/03/05 12:00:01 AM\n'
-                      '83/07/09 5:17:34 PM\n'
-                      '2054/06/20 2:31:45 PM\n'
-                      '00/10/31 11:50:23 AM\n')
-        expected = [datetime.datetime(2014, 1, 11, 0, 0),
-                    datetime.datetime(1976, 3, 5, 0, 0, 1),
-                    datetime.datetime(1983, 7, 9, 17, 17, 34),
-                    datetime.datetime(2054, 6, 20, 14, 31, 45),
-                    datetime.datetime(2000, 10, 31, 11, 50, 23)]
-        self.fd.seek(0)
-        array = mlab.csv2rec(self.fd, names='a', yearfirst=True)
-        assert_array_equal(array['a'].tolist(), expected)
-
-
-class Test_rec2txt(CleanupTestCase):
-    def test_csv2txt_basic(self):
-        # str() calls around field names necessary b/c as of numpy 1.11
-        # dtype doesn't like unicode names (caused by unicode_literals import)
-        a = np.array([(1.0, 2, 'foo', 'bing'),
-                      (2.0, 3, 'bar', 'blah')],
-                     dtype=np.dtype([(str('x'), np.float32),
-                                     (str('y'), np.int8),
-                                     (str('s'), str, 3),
-                                     (str('s2'), str, 4)]))
-        truth = ('       x   y   s     s2\n'
-                 '   1.000   2   foo   bing   \n'
-                 '   2.000   3   bar   blah   ').splitlines()
-        assert mlab.rec2txt(a).splitlines() == truth
-
-
-class Test_window(CleanupTestCase):
-    def setUp(self):
+@pytest.fixture
+def tempcsv():
+    if six.PY3:
+        fd = tempfile.TemporaryFile(suffix='csv', mode="w+", newline='')
+    else:
+        fd = tempfile.TemporaryFile(suffix='csv', mode="wb+")
+    with fd:
+        yield fd
+
+
+def test_recarray_csv_roundtrip(tempcsv):
+    expected = np.recarray((99,),
+                           [(str('x'), float),
+                            (str('y'), float),
+                            (str('t'), float)])
+    # initialising all values: uninitialised memory sometimes produces
+    # floats that do not round-trip to string and back.
+    expected['x'][:] = np.linspace(-1e9, -1, 99)
+    expected['y'][:] = np.linspace(1, 1e9, 99)
+    expected['t'][:] = np.linspace(0, 0.01, 99)
+
+    mlab.rec2csv(expected, tempcsv)
+    tempcsv.seek(0)
+    actual = mlab.csv2rec(tempcsv)
+
+    assert_allclose(expected['x'], actual['x'])
+    assert_allclose(expected['y'], actual['y'])
+    assert_allclose(expected['t'], actual['t'])
```

Comment:
could we just compare the entire arrays at once? Or parametrize the column names if you want more granular testing